byterun/shared_heap.c

```diff
@@ -220,6 +232,62 @@ static pool* pool_find(struct caml_heap_state* local, sizeclass sz) {
   r = local->avail_pools[sz];
   if (r) return r;
 
+/* Haven't managed to find a pool locally, try the global ones */
```

Comment:
Can you briefly explain what this addition is doing and why the earlier code did not have this path?

Reply:
This should be addressed in the opening comment on the PR now.